src/main/java/io/anserini/index/IndexCollection.java

```diff
@@ -183,17 +183,24 @@ public void run() {
 
         int cnt = 0;
         AbstractFileSegment iter = ((FileSegmentProvider) collection).createFileSegment(inputFile);
-        while (iter.hasNext()) {
-          SourceDocument d;
+        while (true) {
+          boolean hasNext = false;
           try {
-            d = iter.next();
+            hasNext = iter.hasNext();
           } catch (NoSuchElementException e1) {
-            continue;
-          } catch (Exception e2) { // TODO: update related counters (#317)
+            break;
+          } catch (Exception e2) {
             LOG.warn("Exception when parsing document: ", e2);
```

Comment:
We are just reading the next document so it should be "Exception when reading document: ", right?